### ISSUE TITLE
Spec definition of gen_udp:open incorrect

### ISSUE THREAD

**Issue by mats9** [erlang/otp#8158]
**Describe the bug**
Type definition of argument Opts for function gen_udp:open is incorrect. `socket:sockaddr_in() | socket:sockaddr_in6()` is missing. This messes up dialyzer analysis, leading to a lot of `Function xxx has no local return`.
**To Reproduce**
Given the following code:
```
open() ->
     Address = {64937,16460,2521,41110,55327,9012,59356,43781},
     Addr = #{family => inet6, port => 8000, addr => Address, scope_id => 2},
     gen_udp:open(0, [binary, {ifaddr, Addr}]).
```
Running typer on above code gives:
```
-spec open() -> none().
```
**Expected behavior**
Output should be:
```
-spec open() -> {'error',atom()} | {'ok',port() | {'$inet',atom(),_}}.
```
**Affected versions**
OTP-26.2.2

**Comment by bmk** [comment]
You are correct, the 'ifaddr' option needed some additional address variants..

**Comment by RaimoNiskanen** [comment]
Merged as 08a613a57bdd277c3b7da8e9cece35291a199400 and released in OTP-26.2.3.

Thank you for reporting this issue!
